tools/integration-tests/tester/framework/config/config.go

```diff
@@ -31,76 +49,116 @@ type GoShimmer struct {
 	DRNG
 }
 
+// NewGoShimmer creates a GoShimmer config initialized with default values.
+func NewGoShimmer() (config GoShimmer) {
+	config = GoShimmer{}
+	fillStructFromDefaultTag(&config)
+	return
+}
+
+// fillStructFromDefaultTag recursively explores the given struct pointer and sets values of fields to the `default` as
+// specified in the struct's tags.
+func fillStructFromDefaultTag(s interface{}) {
```

Comment:
doesn't similar logic exist in hive.go?
don't we have sime duplication?

Reply:
It is indeed similar but not exactly the same as this function is actually filling in the struct with the default values. In hive.go it uses them to create flags.

Reply:
still
shouldn't it be in hive?

Reply:
This struct filling code is very specific to the integration framework itself; and probably there is no general use-case for this that would justify placing this functionality in hive.go.